### ISSUE TITLE
Using antialiasing with RenderWindow + RenderTextures -> Only the 1st RenderTexture is drawn

### ISSUE THREAD

**Issue by danieljpetersen** [SFML/SFML#2630]
Using 2 RenderTextures in conjunction with setting antialiasing on RenderTextures + setting antialiasing on RenderWindow causes the second RenderTexture to not draw correctly, provided a specific order of display() and draw(). 

Here's a brief outline of the steps involved, but see the code example to get a better idea
* Set antialiasing on RenderWindow
* Set antialiasing on 2 RenderTextures
* Call display on first RenderTexture, draw the first RenderTexture to window
* Call display on second RenderTexture, draw second RenderTexture to window
* The first RenderTexture will draw correctly, the second will not draw.

Note that if the two display calls are grouped together before any window->draw(...) calls, the bug does not occur. The order must be like: RenderTexture1->display(), window->draw(renderTexture1); RenderTexture2->display(), window->draw(renderTexture2); 

Both [Bambo](https://github.com/Bambo-Borris) (Windows 11 64 bit with 30 series Nvidia card) and [metaquarx](https://github.com/metaquarx) (Linux, unsure what GPU) have reproduced the bug. I'm on Linux with an AMD laptop. It's been tested and occurs on both 2.6 and 3.0.

Also note that I have not tested more combinations. For example, I have not tested things like setting AA on the window and rendertexture1 but not aa on renderTexture2, nor have I tested things like using more than two render textures.

The textures themselves save to disk as the correct image, it's a matter of rendering them to the screen. Though thinking about it I haven't checked if it actually has AA when you save it to disk.

Here's a reproducable example. Note that you'll need a "texture.png" next to the binary, ideally 800x600. I've attached an image you may want to use after the example. There's also a flag you can set at the top of the code to disable / enable setting AA on the render textures, thus enabling / disabling the issue. 

```

#include <SFML/System.hpp>
#include <SFML/Graphics.hpp>
#include <SFML/Window.hpp>
#include <array>

////////////////////////////////////////////////////////////

// change this to 1 to trigger the bug
#define TRIGGER_THE_BUG 1

int main()
{
    const float screenWidth = 800.0f;
    const float screenHeight = 600.0f;

    sf::RenderWindow window(sf::VideoMode({(unsigned int)screenWidth, (unsigned int)screenHeight}), "Window", sf::Style::Default, sf::ContextSettings(0, 0, 4));
    window.setVerticalSyncEnabled(true);

    sf::Texture beautifulTexture;
    (void)beautifulTexture.loadFromFile("texture.png");
    std::array<sf::RenderTexture, 2> renderTextures {
        sf::RenderTexture(),
        sf::RenderTexture()
    };

    int renderTextureAALevel = 0;
#if TRIGGER_THE_BUG == 1
    renderTextureAALevel = 4;
#endif

    (void)renderTextures[0].create({(unsigned int)screenWidth, (unsigned int)screenHeight}, sf::ContextSettings(0, 0, renderTextureAALevel));
    (void)renderTextures[1].create({(unsigned int)screenWidth, (unsigned int)screenHeight}, sf::ContextSettings(0, 0, renderTextureAALevel));

    auto drawToRenderTextures =[&]() {
        renderTextures[0].clear();
        renderTextures[1].clear();

        sf::Sprite sprite(beautifulTexture);
        renderTextures[0].draw(sprite);

        sprite.setColor(sf::Color::Green); // tint it green so we have some visual distinction between the two renderTextures
        renderTextures[1].draw(sprite);
    };

    std::array<sf::VertexArray, 2> vertexArrays;
    vertexArrays[0].setPrimitiveType(sf::PrimitiveType::Triangles);
    vertexArrays[1].setPrimitiveType(sf::PrimitiveType::Triangles);

    auto makeVertexArrays = [&]() {
        vertexArrays[0].clear();
        vertexArrays[1].clear();

        float xCenter = screenWidth / 2;

        // left half of screen
        vertexArrays[0].append(sf::Vertex({0,0}, {0,0}));
        vertexArrays[0].append(sf::Vertex({xCenter,0}, {xCenter,0}));
        vertexArrays[0].append(sf::Vertex({0,screenHeight}, {0,screenHeight}));

        vertexArrays[0].append(sf::Vertex({0,screenHeight}, {0,screenHeight}));
        vertexArrays[0].append(sf::Vertex({xCenter,0}, {xCenter,0}));
        vertexArrays[0].append(sf::Vertex({xCenter,screenHeight}, {xCenter,screenHeight}));

        // right half of screen
        vertexArrays[1].append(sf::Vertex({xCenter,0}, {xCenter,0}));
        vertexArrays[1].append(sf::Vertex({screenWidth,0}, {screenWidth,0}));
        vertexArrays[1].append(sf::Vertex({xCenter,screenHeight}, {xCenter,screenHeight}));

        vertexArrays[1].append(sf::Vertex({xCenter,screenHeight}, {xCenter,screenHeight}));
        vertexArrays[1].append(sf::Vertex({screenWidth,0}, {screenWidth,0}));
        vertexArrays[1].append(sf::Vertex({screenWidth,screenHeight}, {screenWidth,screenHeight}));
    };

    while (window.isOpen())
    {
        sf::Event event;
        while (window.pollEvent(event))
        {
            if (event.type == sf::Event::Closed)
                window.close();
        }

        window.clear();

        makeVertexArrays();
        drawToRenderTextures();

        // NOTE -- The order of display and draw calls are part of the equation.
        // If We call display() on both render textures and then call draws,
        // there is no bug even with the macro defined to 1.

        renderTextures[0].display();
        window.draw(vertexArrays[0], &renderTextures[0].getTexture());

        renderTextures[1].display();
        window.draw(vertexArrays[1], &renderTextures[1].getTexture());

        /*  EG If the following shit were the code rather than the above, it would work
        renderTextures[0].display();
        renderTextures[1].display();

        window.draw(vertexArrays[0], &renderTextures[0].getTexture());
        window.draw(vertexArrays[1], &renderTextures[1].getTexture());
        //*/

        window.display();
    }

    return 0;
}

```

Test Image you can place next to the binary
![texture](https://github.com/SFML/SFML/assets/5576311/3d9031f2-9f3e-4c7f-b96e-dcb4d69973bf)

Here is the expected output
![correct](https://github.com/SFML/SFML/assets/5576311/20cd77bd-bfad-4e5b-887e-d2e234a9317a)

Here is the buggy output
![buggy](https://github.com/SFML/SFML/assets/5576311/055063a2-552e-4d02-880b-9c435f8aa446)

**Comment by binary1248** [comment]
Here is an even smaller example that breaks:
```cpp
#include <SFML/Graphics.hpp>

int main()
{
    sf::RenderWindow  window(sf::VideoMode({200, 200}), "Test");
    sf::RenderTexture renderTexture;
    (void)renderTexture.create({200, 200}, sf::ContextSettings(0, 0, 1));
    sf::Sprite sprite(renderTexture.getTexture());

    while (window.isOpen())
    {
        sf::Event event;
        while (window.pollEvent(event))
        {
            if (event.type == sf::Event::Closed)
                window.close();
        }

        renderTexture.clear(sf::Color::Green);
        window.clear();
        renderTexture.display();
        window.draw(sprite);
        window.display();
    }
}
```
If everything worked as expected, the window should be filled with green. Instead, it is filled with black.

**Comment by binary1248** [comment]
Should be fixed in #2726.

**Comment by danieljpetersen** [comment]
I just tried it out, that did fix the issue. Thank you!
